pymove/visualization/folium.py

```diff
@@ -0,0 +1,2168 @@
+import folium
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from folium import plugins
+from folium.plugins import FastMarkerCluster, HeatMap, HeatMapWithTime, MarkerCluster
+from tqdm import tqdm_notebook as tqdm
+
+from pymove.preprocessing import filters
+from pymove.utils import constants, distances
+from pymove.utils.constants import (
+    COLORS,
+    COUNT,
+    DATE,
+    DATETIME,
+    DAY,
+    HOUR,
+    LATITUDE,
+    LONGITUDE,
+    PERIOD,
+    SITUATION,
+    STOP,
+    TILES,
+    TRAJ_ID,
+)
+from pymove.utils.datetime import str_to_datetime
+from pymove.utils.mapfolium import add_map_legend
+
+
+def generate_color():
+    """
```

Comment:
Codacy found an issue: [Multi-line docstring summary should start at the first line (D212)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5714604)